site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,190 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Row-Level Deletes
+
+Iceberg supports metadata-based deletion through the `DeleteFiles` interface.
+It allows you to quickly delete a specific file or any file that might match a given expression without the need to read or write any data in the table.
+
+Row-level deletes target more complicated use cases such as general data protection regulation (GDPR).
+Copy-on-write and merge-on-read are two different approaches to handle row-level delete operations. Here are their definitions in Iceberg:
+
+- **copy-on-write**: a delete directly rewrites all the affected data files.
+- **merge-on-read**: delete information is encoded in the form of _delete files_. The table reader can apply all delete information at read time.
```

Comment:
It may just be me, but I strongly prefer the terms **eager** and **lazy** to describe these differences rather than copy-on-write or merge-on-read. Even if we clarify in the definitions here that we mean copy-on-write and merge-on-read, I think it would be better so that we can to the behavior with more descriptive terms.

For example:

> In the eager approach, given a user's delete requirement, the write process will search for all the affected data files and perform a rewrite operation.

I think that makes it really clear what is happening and why.